x/programs/rust/wasmlanche_sdk/src/types.rs

```diff
@@ -105,9 +106,9 @@ impl Argument for i64 {
     }
 }
 
-impl Argument for State {
+impl Argument for Program {
     fn as_bytes(&self) -> Cow<'_, [u8]> {
-        self.program_id.as_bytes()
+        Cow::Owned(self.id().to_be_bytes().to_vec())
```

Comment:
This doesn't make sense...

Reply:
yeah the usage is in the entire file I will dig into it.

Reply:
still working through this.

Reply:
Let me know if you need help. We used `Cow` so that you didn't always have to allocate, but I don't think that's relevant anymore.

Reply:
As we talked about this is as expected with possible change in future followup.